metadata-ingestion/src/datahub/cli/container_cli.py

```diff
@@ -0,0 +1,90 @@
+import logging
+
+import click
+
+from datahub.ingestion.graph.client import get_default_graph
+from datahub.metadata.schema_classes import (
+    GlossaryTermAssociationClass,
+    OwnerClass,
+    OwnershipTypeClass,
+    TagAssociationClass,
+)
+from datahub.specific.dataset import DatasetPatchBuilder
+from datahub.utilities.urns.urn import guess_entity_type
+
+logger = logging.getLogger(__name__)
+
+
+@click.group()
+def container() -> None:
+    """A group of commands to interact with containers in DataHub."""
+    pass
+
+
+def apply_association_to_container(
+    container_urn: str,
+    association_urn: str,
+    association_type: str,
+) -> None:
+    """
+    Common function to apply either tags, terms, or owners to containers and their datasets.
+
+    Args:
+        container_urn: The URN of the container
+        association_urn: The URN of the tag, term, or user to apply
+        association_type: One of 'tag', 'term', or 'owner'
+    """
+    urns = [container_urn]
+    graph = get_default_graph()
+    logger.info(f"Using {graph}")
+    urns.extend(
+        graph.get_urns_by_filter(
+            container=container_urn,
+            batch_size=1000,
+        )
+    )
+
+    for urn in urns:
+        if guess_entity_type(urn) != "dataset":
```

Comment:
you can skip this filter if properly setting `entity_types` in `get_urns_by_filter`

```
    def get_urns_by_filter(
        self,
        *,
        entity_types: Optional[List[str]] = None,
        platform: Optional[str] = None,
        platform_instance: Optional[str] = None,
        env: Optional[str] = None,
        query: Optional[str] = None,
        container: Optional[str] = None,
        ...
```